lib/src/query.dart

```diff
@@ -0,0 +1,766 @@
+import "dart:ffi";
+
+import "box.dart";
+import "store.dart";
+import "common.dart";
+import "bindings/bindings.dart";
+import "bindings/constants.dart";
+import "bindings/flatbuffers.dart";
+import "bindings/helpers.dart";
+import "bindings/structs.dart";
+import "bindings/signatures.dart";
+import "package:ffi/ffi.dart";
+
+// TODO a future or stream(?) should be attachable to the Condition types
+// TODO in case there is an error, or some internal message that should be exposed
+// TODO do async
+/**
+ * The QueryProperty types are responsible
+ * for the operator overloading.
+ *
+ * A QueryBuilder will be constructed,
+ * based on the any / all operations applied.
+ *
+ * When build() is called on the QueryBuilder
+ * a Query object will be created.
+ */
+class QueryProperty {
+  int propertyId; // why hide this if it's exposed in the entity definition, // TODO final / const?
+  int entityId;
+  QueryProperty(this.entityId, this.propertyId);
+
+  QueryCondition isNull() {
+    // the bool serves as a dummy type, to initialize the base type
+    final c = Condition<bool>(ConditionOp._null, null, false);
+    return new QueryCondition(entityId, propertyId, c);
+  }
+
+  QueryCondition notNull() {
+    final c = Condition<int>(ConditionOp._not_null, null, 0);
+    return new QueryCondition(entityId, propertyId, c);
+  }
+}
+
+class QueryStringProperty extends QueryProperty {
+  QueryStringProperty(int entityId, int propertyId) : super(entityId, propertyId);
+
+  static const ConditionType type = ConditionType._string;
```

Comment:
This was a temporary hack to provide sane type mappings. Since adding those params to `Property` I could take user-defined mappings instead.

Actually, to think of it, `OBXPropertyType` could cover these cases.

Reply:
Done.